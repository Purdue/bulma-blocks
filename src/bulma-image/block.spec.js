jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock
} from '@wordpress/e2e-test-utils';

import {
    clickRadio,
    blockStartup,
    selectOption,
    clickElementByText, 
    clickCheckbox
} from '../test-helpers'

const block = {blockTitle: 'Bulma - Image', blockName: 'bulma-blocks/image'}

describe( `ðŸ”¬ ${block.blockTitle} Block`, () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock(block.blockTitle)

        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    describe( 'ðŸ”¬ Block Editor Settings', () => {
        test( 'ðŸ”Ž Media Library Button', async () => {
            await blockStartup(block)
    
            // open media library
            await clickElementByText('button', 'Open Media Library')
    
            // tests that media library does open
            expect(await page.$(`div.media-frame-title`)).not.toBeNull()
        })
    })

    describe( 'ðŸ”¬ Side Panel Settings', () => {
        describe( 'ðŸ”¬ Image Size and Alignment Settings', () => {
            test( 'ðŸ”Ž Aspect Ratio Options', async () => {
                await blockStartup(block)

                // Square 
                await selectOption('Image Aspect Ratio', 'is-square')
                
                let editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-square"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 5:4 
                await selectOption('Image Aspect Ratio', 'is-5by4')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-5by4"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 4:3 
                await selectOption('Image Aspect Ratio', 'is-4by3')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-4by3"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 3by2 
                await selectOption('Image Aspect Ratio', 'is-3by2')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-3by2"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 5by3 
                await selectOption('Image Aspect Ratio', 'is-5by3')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-5by3"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 16by9 
                await selectOption('Image Aspect Ratio', 'is-16by9')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-16by9"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 2by1 
                await selectOption('Image Aspect Ratio', 'is-2by1')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-2by1"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 3by1 
                await selectOption('Image Aspect Ratio', 'is-3by1')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-3by1"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 4by5 
                await selectOption('Image Aspect Ratio', 'is-4by5')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-4by5"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 3by4 
                await selectOption('Image Aspect Ratio', 'is-3by4')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-3by4"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 2by3 
                await selectOption('Image Aspect Ratio', 'is-2by3')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-2by3"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 3by5 
                await selectOption('Image Aspect Ratio', 'is-3by5')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-3by5"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 9by16 
                await selectOption('Image Aspect Ratio', 'is-9by16')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-9by16"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 1by2 
                await selectOption('Image Aspect Ratio', 'is-1by2')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-1by2"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // 1by3 
                await selectOption('Image Aspect Ratio', 'is-1by3')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-1by3"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // auto 
                await selectOption('Image Aspect Ratio', '')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"size":"is-square"')).toBe(false)
                expect( editedContent.includes('"size":"is-5by4"')).toBe(false)
                expect( editedContent.includes('"size":"is-4by3"')).toBe(false)
                expect( editedContent.includes('"size":"is-3by2"')).toBe(false)
                expect( editedContent.includes('"size":"is-5by3"')).toBe(false)
                expect( editedContent.includes('"size":"is-16by9"')).toBe(false)
                expect( editedContent.includes('"size":"is-2by1"')).toBe(false)
                expect( editedContent.includes('"size":"is-3by1"')).toBe(false)
                expect( editedContent.includes('"size":"is-4by5"')).toBe(false)
                expect( editedContent.includes('"size":"is-3by4"')).toBe(false)
                expect( editedContent.includes('"size":"is-2by3"')).toBe(false)
                expect( editedContent.includes('"size":"is-3by5"')).toBe(false)
                expect( editedContent.includes('"size":"is-9by16"')).toBe(false)
                expect( editedContent.includes('"size":"is-1by2"')).toBe(false)
                expect( editedContent.includes('"size":"is-1by3"')).toBe(false)
                expect( editedContent ).toMatchSnapshot()
            })

            test( 'ðŸ”Ž Image Alignment Options', async () => {
                await blockStartup(block)

                // center 
                await selectOption('Alignment', 'center')
                
                let editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"align":"center"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // left 
                await selectOption('Alignment', 'left')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"align":"left"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // right 
                await selectOption('Alignment', 'right')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"align":"right"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                // full 
                await selectOption('Alignment', '')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"align":"center"')).toBe(false)
                expect( editedContent.includes('"align":"left"')).toBe(false)
                expect( editedContent.includes('"align":"right"')).toBe(false)
                expect( editedContent ).toMatchSnapshot()
            })

            test( 'ðŸ”Ž Mobile Stretch Checkbox', async () => {
                await blockStartup(block)

                await clickCheckbox('Full Width on Mobile?')

                let editedContent = await getEditedPostContent()

                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"fullWidthMo":true')).toBe(true)
            })
            
            test( 'ðŸ”Ž Height and Width Options', async () => {
                await blockStartup(block)

                const typeString = "400"

                // test width textbox
                await page.focus('#imgWidth')
                await page.keyboard.type(typeString, {delay: 10})
                await page.keyboard.press('Enter')

                let editedContent = await getEditedPostContent()

                expect( editedContent).toMatchSnapshot()
                expect( editedContent.includes(`"imgW":"${typeString}"`)).toBe(true)

                // test height textbox
                await page.focus('#imgHeight')
                await page.keyboard.type(typeString, {delay: 10})
                await page.keyboard.press('Enter')

                editedContent = await getEditedPostContent()

                expect( editedContent).toMatchSnapshot()
                expect( editedContent.includes(`"imgH":"${typeString}"`)).toBe(true)
            })
        })
    
        describe( 'ðŸ”¬ Image Alt Text and Caption Settings', () => {
            test( 'ðŸ”Ž Alt Text Textarea', async () => {
                await blockStartup(block)

                const typeString = "Test Alt Text"

                await clickElementByText('label', 'Alt Text')
                await page.keyboard.type(typeString, {delay: 10})

                let editedContent = await getEditedPostContent()

                expect( editedContent).toMatchSnapshot()
                expect( editedContent.includes(`"altText":"${typeString}"`)).toBe(true)
            })
            test( 'ðŸ”Ž Image Caption Textarea', async () => {
                await blockStartup(block)

                const typeString = "Test Image Caption"

                await clickElementByText('label', 'Image Caption')
                await page.keyboard.type(typeString, {delay: 10})

                let editedContent = await getEditedPostContent()

                expect( editedContent).toMatchSnapshot()
                expect( editedContent.includes(`"caption":"${typeString}"`)).toBe(true)
            })
            test( 'ðŸ”Ž Image Photo Credit Textbox', async () => {
                await blockStartup(block)

                const typeString = "Test Image Caption Credit"

                await clickElementByText('label', 'Caption - Photo Credit')
                await page.keyboard.type(typeString, {delay: 10})
                await page.keyboard.press('Enter')

                let editedContent = await getEditedPostContent()

                expect( editedContent).toMatchSnapshot()
                expect( editedContent.includes(`"captionCredit":"${typeString}"`)).toBe(true)
            })
        })
    })

})