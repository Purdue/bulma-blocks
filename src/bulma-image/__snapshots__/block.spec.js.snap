// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Bulma - Image Block 🔎 Block should be available. 1`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<figure class=\\"image   full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Alt Text and Caption Settings 🔎 Alt Text Textarea 1`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"Test Alt Text\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<figure class=\\"image   full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"Test Alt Text\\" src=\\"\\" class=\\"\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Alt Text and Caption Settings 🔎 Image Caption Textarea 1`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"Test Image Caption\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<figure class=\\"image   full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/><div class=\\"bulma-blocks-image__caption\\"><p>Test Image Caption</p></div></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Alt Text and Caption Settings 🔎 Image Photo Credit Textbox 1`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"Test Image Caption Credit\\",\\"align\\":\\"\\"} -->
<figure class=\\"image   full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 1`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-square\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-square full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 2`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-5by4\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-5by4 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 3`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-4by3\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-4by3 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 4`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-3by2\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-3by2 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 5`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-5by3\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-5by3 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 6`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-16by9\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-16by9 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 7`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-2by1\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-2by1 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 8`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-3by1\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-3by1 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 9`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-4by5\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-4by5 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 10`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-3by4\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-3by4 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 11`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-2by3\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-2by3 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 12`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-3by5\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-3by5 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 13`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-9by16\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-9by16 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 14`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-1by2\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-1by2 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 15`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"is-1by3\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<div><figure class=\\"image is-1by3 full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure></div>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Aspect Ratio Options 16`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<figure class=\\"image   full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Height and Width Options 1`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"400\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<figure class=\\"image   full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Height and Width Options 2`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"400\\",\\"imgH\\":\\"400\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<figure class=\\"image   full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Image Alignment Options 1`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"center\\"} -->
<figure class=\\"image  center  \\" style=\\"margin-bottom:0.5rem !important\\"><img alt=\\"\\" src=\\"\\" style=\\"max-width:px;max-height:px\\" class=\\"\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Image Alignment Options 2`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"left\\"} -->
<figure class=\\"image  left  \\" style=\\"margin-bottom:0.5rem !important\\"><img alt=\\"\\" src=\\"\\" style=\\"max-width:px;max-height:px\\" class=\\"\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Image Alignment Options 3`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"right\\"} -->
<figure class=\\"image  right  \\" style=\\"margin-bottom:0.5rem !important\\"><img alt=\\"\\" src=\\"\\" style=\\"max-width:px;max-height:px\\" class=\\"\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Image Alignment Options 4`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\"} -->
<figure class=\\"image   full \\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;

exports[`🔬 Bulma - Image Block 🔬 Side Panel Settings 🔬 Image Size and Alignment Settings 🔎 Mobile Stretch Checkbox 1`] = `
"<!-- wp:bulma-blocks/image {\\"size\\":\\"\\",\\"imgUrl\\":\\"\\",\\"imgW\\":\\"\\",\\"imgH\\":\\"\\",\\"altText\\":\\"\\",\\"caption\\":\\"\\",\\"captionCredit\\":\\"\\",\\"align\\":\\"\\",\\"fullWidthMo\\":true} -->
<figure class=\\"image   full full-width-mo\\" style=\\"margin-bottom:3rem !important\\"><img alt=\\"\\" src=\\"\\" class=\\"full-width-mo\\"/></figure>
<!-- /wp:bulma-blocks/image -->"
`;
